cardano-node/Cardano/Node/CLI.hs

```diff
@@ -61,30 +71,47 @@ data Protocol =
   | MockPBFT
   | RealPBFT
 
+-- | Tracing-related constraints for monitoring purposes.
+--
+-- When you need a 'Show' or 'Condense' instance for more types, just add the
+-- appropriate constraint here. There's no need to modify the consensus
+-- code-base, unless the corresponding instance is missing.
+type TraceConstraints blk =
+    ( Condense blk
+    , Condense [blk]
+    , Condense (ChainHash blk)
+    , Condense (Header blk)
+    , Condense (HeaderHash blk)
+    , Condense (GenTx blk)
+    , Show (ApplyTxErr blk)
+    , Show (GenTx blk)
+    )
+
 data SomeProtocol where
-  SomeProtocol :: RunDemo blk => DemoProtocol blk -> SomeProtocol
+  SomeProtocol :: (RunDemo blk, TraceConstraints blk)
+               => Consensus.Protocol blk -> SomeProtocol
 
 fromProtocol :: Protocol -> IO SomeProtocol
 fromProtocol BFT =
-    case runDemo p of
+    case Consensus.runProtocol p of
       Dict -> return $ SomeProtocol p
   where
-    p = DemoBFT defaultSecurityParam
+    p = ProtocolMockBFT defaultSecurityParam
 fromProtocol Praos =
-    case runDemo p of
+    case Consensus.runProtocol p of
       Dict -> return $ SomeProtocol p
   where
-    p = DemoPraos defaultDemoPraosParams
+    p = ProtocolMockPraos defaultDemoPraosParams
 fromProtocol MockPBFT =
-    case runDemo p of
+    case Consensus.runProtocol p of
       Dict -> return $ SomeProtocol p
   where
-    p = DemoMockPBFT defaultDemoPBftParams
+    p = ProtocolMockPBFT defaultDemoPBftParams
 fromProtocol RealPBFT =
-    case runDemo p of
-      Dict -> return $ SomeProtocol p
+    case Consensus.runProtocol p of
+      Dict -> return $ undefined -- SomeProtocol p
```

Comment:
@nc6 I get:
```
[-Wdeferred-type-errors]
    • Could not deduce (reflection-2.1.4:Data.Reflection.Given
                          cardano-ledger-0.1.0.0:Cardano.Chain.Block.Header.HeaderHash)
        arising from a use of ‘SomeProtocol’
      from the context: Ouroboros.Consensus.Node.Run.Abstract.RunNode
                          (Ouroboros.Consensus.Ledger.Byron.ByronBlock ByronConfig)
        bound by a pattern with constructor:
                   Dict :: forall (a :: Constraint). a => Dict a,
                 in a case alternative
    • In the second argument of ‘($)’, namely ‘SomeProtocol p’
      In the expression: return $ SomeProtocol p
      In a case alternative: Dict -> return $ SomeProtocol p
```

So `ByronGiven`. I don't know why I get this error now and not before.

Reply:
@nc6 Any progress on this?